File: benches/python/factorial.py
#!/usr/bin/env python3
"""Factorial benchmark for Python"""

def fact(n: int) -> int:
    if n <= 1:
        return 1
    return n * fact(n - 1)

def main():
    import sys
    n = int(sys.argv[1]) if len(sys.argv) > 1 else 15
    result = fact(n)
    print(result)

if __name__ == "__main__":
    main()

File: benches/rust/factorial.rs
// Factorial benchmark for Rust

fn fact(n: i64) -> i64 {
    if n <= 1 {
        return 1;
    }
    n * fact(n - 1)
}

fn main() {
    let n = std::env::args()
        .nth(1)
        .and_then(|s| s.parse().ok())
        .unwrap_or(15);
    let result = fact(n);
    println!("{}", result);
}

